zoom.sh

```diff
@@ -1,22 +1,26 @@
 #!/usr/bin/env bash
 
+CONF_FILE=$HOME/.zoom/config/zoomus.conf
+
 # Check if the current desktop is using Wayland
 if [[ "$XDG_SESSION_TYPE" == "wayland" ]]; then
+    QPA=wayland
 
-	# Check if the line 'enableWaylandShare' is set to 'true' in $HOME/.var/app/us.zoom.Zoom/config/zoomus.conf
-	if ! grep -q enableWaylandShare=true "$HOME/.var/app/$FLATPAK_ID/config/zoomus.conf"; then
-
-		# Replace enableWaylandShare=false to enableWaylandShare=true
-		sed -i 's/enableWaylandShare\=false/enableWaylandShare\=true/' ~/.var/app/$FLATPAK_ID/config/zoomus.conf
+    # Check if the line 'enableWaylandShare' is set to 'true' in $HOME/.var/app/us.zoom.Zoom/config/zoomus.conf
+    if ! grep -q enableWaylandShare=true "$CONF_FILE"; then
 
-		# Recheck if the line 'enableWaylandShare' is set to true in $HOME/.var/app/us.zoom.Zoom/config/zoomus.conf
-		if ! grep -q enableWaylandShare=true "$HOME/.var/app/$FLATPAK_ID/config/zoomus.conf"; then
+        # Replace enableWaylandShare=false to enableWaylandShare=true
+        sed -i 's/enableWaylandShare\=false/enableWaylandShare\=true/' $CONF_FILE
 
-			# If not, create a GTK dialog that says the string inside '--text'
-        	       	zenity --error --text="Wayland screen sharing is not yet enabled. Please restart Zoom for it to automatically enable, or manually change the value of \"enableWaylandShare\" to \"true\" in \"$HOME/.var/app/$FLATPAK_ID/config/zoomus.conf\"."
+        # Recheck if the line 'enableWaylandShare' is set to true in $HOME/.var/app/us.zoom.Zoom/config/zoomus.conf
+        if ! grep -q enableWaylandShare=true "$CONF_FILE"; then
 
-		fi
-       	fi
+            # If not, create a GTK dialog that says the string inside '--text'
+            zenity --error --text="Wayland screen sharing is not yet enabled. Please restart Zoom for it to automatically enable, or manually change the value of \"enableWaylandShare\" to \"true\" in \"$CONF_FILE\"."
+        fi
+    fi
+else 
+    QPA=xcb
 fi
 
```

Comment:
```suggestion

if [ -z "$DISPLAY" ] && [ -n "$WAYLAND_DISPLAY" ]; then
    QPA=wayland
fi
```

True if actively running in Wayland and Wayland Flatpak socket is enabled. Together with the below change (don't pass Wayland socket by default) means Zoom will run in Xwayland by default, and if someone enables the Wayland Flatpak socket (and disables X11 socket/enables fallback X11) Zoom will run in Native Wayland mode.